internal/server/validation.go

```diff
@@ -28,6 +30,12 @@ func (s *Server) Validate() error {
 	err := valid.Struct(s)
 	valerrs, ok := err.(validator.ValidationErrors)
 
+	if s.CustomNotFoundPage != "" {
+		if _, err := os.Stat(s.CustomNotFoundPage); errors.Is(err, os.ErrNotExist) {
+			return err
+		}
```

Comment:
I think this goes the other way: we want to know if the file doesn't exist, and we also want to know if it couldn't be stat'd at the beginning of the server. Not being able to stat it might mean something happened to it.
```suggestion
		if _, err := os.Stat(s.CustomNotFoundPage); err != nil {
			if errors.Is(err, os.ErrNotExist) {
				return fmt.Errorf("custom not found file %q does not exist", s.CustomNotFoundPage)
			}
			
			return err
		}
```